gpytorch/lazy/interpolated_lazy_tensor.py

```diff
@@ -386,6 +386,13 @@ def diag(self):
         else:
             return super(InterpolatedLazyTensor, self).diag()
 
+    def double(self, device_id=None):
+        # We need to ensure that the indices remain integers.
+        new_lt = super().double(device_id=device_id)
+        new_lt.left_interp_indices = new_lt.left_interp_indices.type(torch.int64)
+        new_lt.right_interp_indices = new_lt.right_interp_indices.type(torch.int64)
```

Comment:
Would we ever want to move ints to a FP dtype? Otherwise we can just not move those over in the `float` and `double` calls (could add an optional `include_int` kwarg if we really wanted to allow forcing that). Then we wouldn't have to do this check here (which is likely a gotcha for new types).

Reply:
I don't think we'd ever want to allow these to be non-integer valued due to the explicit indexiing that `InterpolatedLazyTensor` uses. Even `.half/float/double()` etc is really a misnomer because `InterpolatedLazyTensor` will include two different tensor types by default (integers and floating point values).

Reply:
Not passing these over to double precision integers also produces errors as well, so this type of type hack has to be performed in my opinion.